prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -426,11 +426,22 @@ static void server_sock_accept(struct util_wait *wait,
 
 	sock = accept(pep->sock, NULL, 0);
 	if (sock < 0) {
-		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
-			"accept error: %d\n", ofi_sockerr());
+		if (!OFI_SOCK_TRY_ACCEPT_AGAIN(ofi_sockerr())) {
+			FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+				"accept error: %d\n", ofi_sockerr());
+		}
 		return;
 	}
 
+	/* Make sure the new socket doesn't inherit the non-blocking state
+	 * from the PEP socket (default behavior with BSD and OSX). */
+	ret = fi_fd_block(sock);
+	if (ret) {
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+			"failed to set socket to blocking\n");
+		goto err1;
+	}
```

Comment:
It's unfortunate that the tcp provider requires blocking send/recv calls to exchange the CM setup messages.  The send may not be too bad, assuming that a newly connected socket is basically always ready to send at least 1 mtu worth of data.  But the blocking receive looks like it could cause issues by hanging the application thread for long periods of time.